controller/api/destination/server.go

```diff
@@ -264,17 +270,26 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 /// util ///
 ////////////
 
-func nsFromToken(token string) string {
-	// ns:<namespace>
-	parts := strings.Split(token, ":")
-	if len(parts) == 2 && parts[0] == "ns" {
-		return parts[1]
+type contextToken struct {
+	Ns       string `json:"ns,omitempty"`
+	NodeName string `json:"nodeName,omitempty"`
+}
+
+func parseContextToken(token string) (contextToken, error) {
+	ctxToken := contextToken{}
+	if err := json.Unmarshal([]byte(token), &ctxToken); err != nil {
+		// if json is invalid, means token can have ns:<namespace> form
+		parts := strings.Split(token, ":")
+		if len(parts) == 2 && parts[0] == "ns" {
+			ctxToken.Ns = parts[1]
+		}
+		return ctxToken, err
```

Comment:
I think we want to return nil for the second argument here.  We failed to parse as json, but successfully parsed in the old format.

Reply:
In fact, under that framing, this method never returns an error and can return just one return argument.